// toelfdb/lib/dbserver.js
var DBHouse = require('dbhouse');
var BSON = require('mongodb').BSONNative;

var TOEFL = null;

var DBserver = module.exports = function(toefldb) {
	this.toefldb = TOEFLDB = toefldb;
	this.dbName = 'toefl_test';
	this.dbcollectionName = 'english';
    //console.log(toefldb.database);
	this.db = new DBHouse.Database(toefldb.database);
};

/*=====================================*/
/*
/* function name: quiz
/* Description: search db content 
/* Inputs : limit, sort, filter, offset, collection_name, id
/* Defaults : use db as devtof, use collection as default
/* Returns : data
/*
/*=====================================*/

DBserver.prototype.quiz = function (req, res) {
	var self = this;
    //console.log("(lib)DBserver: quiz is connected");
    console.log("skip: " + req.params.skip + ";  limit: " + req.params.limit );
    //res.send("skip: " + req.params.skip + "\nlimit: " + req.params.limit );
    self.db
        .open('toelf_test')
        .collection('english')
        .where({})
        .skip(req.params.skip)
        .limit(req.params.limit)
        .order('_id', 1)
        .query(function(err, data) {
            if (err) {
                console.log("(lib)DBserver: Error. " + err);
                return res.json({ 
                    msg : err
                });
            }
            
            //console.log("(lib)DBserver: data. ");
            //console.log(data);
            var result = new Array();
			for ( var i = 0; i < data.length; i++ ) {
				var obj = new Object();
				obj['_id'] = data[i]['_id'];
				obj['word'] = data[i]['word'];
				obj['chinese'] = data[i]['chinese']['word'];
				obj['provider'] = data[i]['chinese']['provider'];

				result.push(obj);
			}
			return res.json({
                quiz : result
			});
        });
};
